patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch

```diff
@@ -9,7 +9,27 @@
                      {
                          this.field_146015_k = false;
                          this.field_174909_f.clear();
-@@ -181,7 +181,7 @@
+                         break;
+                     }
+ 
+-                    tileentitybeacon$beamsegment.func_177262_a();
+-                    continue;
++                    // Allow any block to change the beacon beam colour
++                    float[] colour = iblockstate.func_177230_c().getBeaconColour(field_145850_b, iblockstate);
++                    if(colour == null || colour.length != 3){
++                        tileentitybeacon$beamsegment.func_177262_a();
++                        continue;
++                    }
++                    afloat = colour;
++                }else {
++                    afloat = EntitySheep.func_175513_a((EnumDyeColor) iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
+                 }
+-
+-                afloat = EntitySheep.func_175513_a((EnumDyeColor)iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
+             }
```

Comment:
Oh, I didn't mean to. Does that effect anything?

On Aug 4, 2016 1:29 PM, "Henry Loenwind" <notifications@github.com> wrote:

> In patches/minecraft/net/minecraft/tileentity/TileEntityBeacon.java.patch
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159#discussion_r73577180>
> :
>
> > +
> > +-                    tileentitybeacon$beamsegment.func_177262_a();
> > +-                    continue;
> > ++                    // Allow any block to change the beacon beam colour
> > ++                    float[] colour = iblockstate.func_177230_c().getBeaconColour(field_145850_b, iblockstate);
> > ++                    if(colour == null || colour.length != 3){
> > ++                        tileentitybeacon$beamsegment.func_177262_a();
> > ++                        continue;
> > ++                    }
> > ++                    afloat = colour;
> > ++                }else {
> > ++                    afloat = EntitySheep.func_175513_a((EnumDyeColor) iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
> > +                 }
> > +-
> > +-                afloat = EntitySheep.func_175513_a((EnumDyeColor)iblockstate.func_177229_b(BlockStainedGlassPane.field_176245_a));
> > +             }
>
> He meant, why do you edit the line? There is no change to anything but
> whitespace. You should leave it alone so your patch would only add liness
> before it and a line with a single "}" after it.
>
> Same for the two lines above. Don't change the indent.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MinecraftForge/MinecraftForge/pull/3159/files/9b5bf699943a12d1e97b3cbae6d96950eb91402e#r73577180>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AFYYWV2IQ0yDm_XSal644sziVz_tYSYdks5qci9zgaJpZM4Jb-cJ>
> .
>